python/tests/expressions/test_expressions_base.py

```diff
@@ -15,9 +15,42 @@
 # specific language governing permissions and limitations
 # under the License.
 
+import uuid
+from decimal import Decimal
+from typing import Any, Union
+
 import pytest
 
 from iceberg.expressions import base
+from iceberg.types import (
+    BinaryType,
+    BooleanType,
+    DecimalType,
+    FloatType,
+    IntegerType,
+    StringType,
+    UUIDType,
+)
+
+
+class FooStruct:
```

Comment:
This implements `StructProtocol` right?

Reply:
Yes I didn't subclass it since it's not required but it would be more helpful to include it here. I'll update this 👍

Reply:
Actually, I think it's a good idea not to include it since that's why we used a Protocol, right?

Reply:
Ok I switched it back 👍